test/IntelFPGAMemoryAttributesForStruct.ll

```diff
@@ -8,246 +169,476 @@
-; CHECK-LLVM: [[STR2:@[0-9_.]+]] = {{.*}}{register:1}
-; CHECK-LLVM: [[STR3:@[0-9_.]+]] = {{.*}}{memory:MLAB}
-; CHECK-LLVM: [[STR4:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{bankwidth:8}
-; CHECK-LLVM: [[STR5:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{private_copies:4}
-; CHECK-LLVM: [[STR6:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{pump:1}
-; CHECK-LLVM: [[STR7:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{pump:2}
-; CHECK-LLVM: [[STR8:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{merge:foobar:width}
-; CHECK-LLVM: [[STR9:@[0-9_.]+]] = {{.*}}{max_replicates:4}
-; CHECK-LLVM: [[STR10:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{simple_dual_port:1}
-; CHECK-LLVM: [[STR12:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{numbanks:8}{bank_bits:42,41,40}
-; CHECK-LLVM: [[STR11:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{numbanks:2}
-@.str = private unnamed_addr constant [29 x i8] c"{memory:DEFAULT}{numbanks:4}\00", section "llvm.metadata"
-@.str.1 = private unnamed_addr constant [16 x i8] c"test_struct.cpp\00", section "llvm.metadata"
-@.str.2 = private unnamed_addr constant [13 x i8] c"{register:1}\00", section "llvm.metadata"
-@.str.3 = private unnamed_addr constant [14 x i8] c"{memory:MLAB}\00", section "llvm.metadata"
-@.str.4 = private unnamed_addr constant [30 x i8] c"{memory:DEFAULT}{bankwidth:8}\00", section "llvm.metadata"
-@.str.5 = private unnamed_addr constant [35 x i8] c"{memory:DEFAULT}{private_copies:4}\00", section "llvm.metadata"
-@.str.6 = private unnamed_addr constant [25 x i8] c"{memory:DEFAULT}{pump:1}\00", section "llvm.metadata"
-@.str.7 = private unnamed_addr constant [25 x i8] c"{memory:DEFAULT}{pump:2}\00", section "llvm.metadata"
-@.str.8 = private unnamed_addr constant [37 x i8] c"{memory:DEFAULT}{merge:foobar:width}\00", section "llvm.metadata"
-@.str.9 = private unnamed_addr constant [19 x i8] c"{max_replicates:4}\00", section "llvm.metadata"
-@.str.10 = private unnamed_addr constant [37 x i8] c"{memory:DEFAULT}{simple_dual_port:1}\00", section "llvm.metadata"
-@.str.11 = private unnamed_addr constant [29 x i8] c"{memory:DEFAULT}{numbanks:2}\00", section "llvm.metadata"
-@.str.12 = private unnamed_addr constant [49 x i8] c"{memory:DEFAULT}{numbanks:8}{bank_bits:42,41,40}\00", section "llvm.metadata"
-
-; Function Attrs: nounwind
+%struct.numbanks_st = type { i32 }
+%struct.templ_numbanks_st = type { i32 }
+%struct.register_st = type { i32 }
+%struct.memory_st = type { i32 }
+%struct.bankwidth_st = type { i32 }
+%struct.templ_bankwidth_st = type { i32 }
+%struct.private_copies_st = type { i32 }
+%struct.templ_private_copies_st = type { i32 }
+%struct.singlepump_st = type { i32 }
+%struct.doublepump_st = type { i32 }
+%struct.merge_st = type { i32 }
+%struct.max_replicates_st = type { i32 }
+%struct.templ_max_replicates_st = type { i32 }
+%struct.simple_dual_port_st = type { i32 }
+%struct.bank_bits_st = type { i32 }
+%struct.templ_bank_bits_st = type { i32 }
+%struct.state = type { [8 x i32] }
+
+; CHECK-LLVM: [[STR01:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{numbanks:4}
```

Comment:
Could `STR`/`FIELD` variables have some short-hand attribute names as suffixes instead of numbers?

Reply:
Done